pandas/core/arrays/integer.py

```diff
@@ -173,8 +173,11 @@ def coerce_to_array(values, dtype, mask=None, copy=False):
     values = np.array(values, copy=copy)
     if is_object_dtype(values):
         inferred_type = lib.infer_dtype(values)
-        if inferred_type not in ['floating', 'integer',
-                                 'mixed-integer', 'mixed-integer-float']:
+        if inferred_type is 'mixed' and isna(values).any():
```

Comment:
Shoudn't this `any` be `all` ? 
Because now this allows something like this: 

```
In [14]: pd.core.arrays.integer_array([np.nan, {'a':1}])
Out[14]: IntegerArray([nan, nan], dtype='Int64')
```

A cornercase, but basically any python object that is not a datetime or string (something that numpy will convert to an object array).

Reply:
good catch, yep need to have all here